### PR TITLE
feat: dynamic timebounds for batch posting

### PR DESCRIPTION
**Pull request by dewanshparashar**
Currently, our batch poster monitors for a fixed duration of 4 hours if a batch has not been posted on parent-chain. With this PR, we now make this range dynamic because it depends on the `maxTimeVariation` set for the chain.

Closes FS-719

### CODE REVIEW

File: packages/batch-poster-monitor/chains.ts [hunk 1]
```diff
@@ -11,8 +12,9 @@ import {
 } from 'viem/chains'
 
 // Defaults
-export const DEFAULT_TIMESPAN_SECONDS = 60 * 60 * 24 // 24 hours : we don't care about txs older than 24 hours
-export const DEFAULT_BATCH_POSTING_DELAY_SECONDS = 60 * 60 * 4 // 4 hours
+export const MAX_TIMEBOUNDS_SECONDS = 60 * 60 * 24 // 24 hours : we don't care about txs older than 24 hours
```
Comment (dewanshparashar) [inline packages/batch-poster-monitor/chains.ts:15]: Updated variable names

File: packages/batch-poster-monitor/index.ts [hunk 2]
```diff
@@ -348,46 +397,54 @@ const monitorBatchPoster = async (childChainInformation: ChainInfo) => {
...
   const latestChildChainBlockNumber = await childChainClient.getBlockNumber()
 
-  if (!sequencerInboxLogs || sequencerInboxLogs.length === 0) {
-    // No SequencerInboxLog in the last 24 hours (time hardcoded in getDefaultBlockRange)
-    // We compare the "latest" and "safe" blocks, and check if any of the pending blocks contain any user-transactions
-    // NOTE: another way of verifying this might be to check the timestamp of the last block in the childChain chain to verify more or less if it should have been posted
-    const latestChildChainSafeBlock = await childChainClient.getBlock({
-      blockTag: 'safe',
-    })
+  const latestChildChainSafeBlock = await childChainClient.getBlock({
```
Comment (TucksonDev) [inline packages/batch-poster-monitor/index.ts:400]: I think most of this code can be inside the `if (!sequencerInboxLogs...)`. That way we can save multiple RPC calls when there are logs (which hopefully will be the common case)

File: packages/batch-poster-monitor/index.ts [hunk 3]
```diff
@@ -348,46 +397,54 @@ const monitorBatchPoster = async (childChainInformation: ChainInfo) => {
...
+    // if alert situation
+    if (batchPostingBacklog) {
       alertsForChildChain.push(
         `No batch has been posted in the last ${
-          DEFAULT_TIMESPAN_SECONDS / 60 / 60
+          MAX_TIMEBOUNDS_SECONDS / 60 / 60
         } hours, and last block number (${latestChildChainBlockNumber}) is greater than the last safe block number (${
           latestChildChainSafeBlock.number
-        })`
+        }). Atleast 1 block must be posted every ${
```
Comment (TucksonDev) [inline packages/batch-poster-monitor/index.ts:430]:
```suggestion
        }). At least 1 batch is expected to be posted every ${
```
Smol nitpick for accuracy in text

File: packages/batch-poster-monitor/index.ts [hunk 4]
```diff
@@ -418,14 +475,16 @@ const monitorBatchPoster = async (childChainInformation: ChainInfo) => {
...
-    secondsSinceLastBatchPoster > DEFAULT_BATCH_POSTING_DELAY_SECONDS
+    secondsSinceLastBatchPoster > BigInt(batchPostingTimeBounds)
   ) {
     alertsForChildChain.push(
       `Last batch was posted ${
         secondsSinceLastBatchPoster / 60n / 60n
       } hours and ${
         (secondsSinceLastBatchPoster / 60n) % 60n
-      } mins ago, and there's a backlog of ${batchPosterBacklog} blocks in the chain`
+      } mins ago, and there's a backlog of ${batchPosterBacklog} blocks in the chain. At least 1 batch must be posted every ${
```
Comment (TucksonDev) [inline packages/batch-poster-monitor/index.ts:485]:
```suggestion
      } mins ago, and there's a backlog of ${batchPosterBacklog} blocks in the chain. At least 1 batch is expected to be posted every ${
```
Smol nitpick

**Review by TucksonDev — CHANGES_REQUESTED** [review]
Looks good! Only requesting one change to save some RPC calls in the happy case.

File: packages/batch-poster-monitor/index.ts [hunk 5]
```diff
@@ -267,6 +275,47 @@ const checkForUserTransactionBlocks = async ({
...
+    console.log({
+      delayBlocks,
+      delaySeconds,
+      batchPostingTimeBounds,
+    })
+  } catch (_) {
+    // no-op, use the fallback value
+  }
+
+  // formula : min(50% of x , max(1h, x - 18h))
```
Comment (TucksonDev) [inline packages/batch-poster-monitor/index.ts:311]: The constant seems to be 9h instead of 18h

Comment (dewanshparashar) [reply packages/batch-poster-monitor/index.ts]: Yes I was adjusting the buffer values to get a reasonable timebound. 9 seems to be working better than 18, but we might need to change it to 6 as well. Will discuss it with the broader team.

File: packages/batch-poster-monitor/index.ts (hunk 4 above)
Comment (dewanshparashar) [reply packages/batch-poster-monitor/index.ts]: Updated

File: packages/batch-poster-monitor/index.ts (hunk 3 above)
Comment (dewanshparashar) [reply packages/batch-poster-monitor/index.ts]: Updated

File: packages/batch-poster-monitor/index.ts (hunk 2 above)
Comment (dewanshparashar) [reply packages/batch-poster-monitor/index.ts]: Updated

**Review by TucksonDev — APPROVED** [review]
LGTM !

**Pull request merged**